---
title: "Using the MiniMax Statistic"
author: "Gabriel Odom and Lily Wang"
date: "8/29/2021"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Using the MiniMax Statistic}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  message = FALSE,
  comment = "#>"
)
```



# Introduction

This vignette will give a brief overview of the practical problem that motivated us to work on this project, some mathematical details about our method, and a walkthrough of a real data analysis example.

```{r}
library(tidyverse)
library(pathwayMultiomics)
```

------------------------------------------------------------------------

</br>



# Motivating Example
Some time ago, our research team attempted to use multi-omics tools to answer a few questions concerning colorectal cancer. While the tools were often well-documented and had nice statistical properties, we consistently ran into one of the following two constraints:

-   Multi-Omics tools must match across platforms *by samples*, or
-   Multi-Omics tools must match across platforms *by features* (e.g. genes).


## Joining data platforms
In many cancer data sets, we could usually find a nice overlapping set of samples between two platforms of data (for instance, copy number variation and DNA methylation). However, as soon as we added in a third or fourth platform (such as gene and/or protein expression), the intersection of the samples recorded with these new platforms rapidly approached the empty set. We could start with 600+ samples, and we’d barely have 50 samples by the time we added a third platform and only a handful of samples (or even none at all) if we added a fourth platform.

If instead, we tried to match on features, we would run into a different set of issues. What is the best way to map from the “tallest” data, site- or probe-level data (as in DNA methylation, SNP, or RNA editing levels) to the “shortest” data, such as expressions of small sets of individual proteins? Using a feature-matching strategy would require us to discard many relevant biological features simply because they were not recorded in the most restricted data.


## A Pathway-based solution
Rather than using only the intersection of the samples or only the intersection of the genomic features, we proposed to match across data sets by data summaries of biological pathway *p*-values. For example, consider omics data measured on three data platforms for colorectal adenocarcinoma: copy number variation, DNA methylation, and protein expression. These three data sets may share samples, but that isn’t required. 

Our first step is to statistically assess the activity of a set of biological pathways in each genomic dataset independently and then match the pathway *p*-values across the three datasets. The resulting summary data would have one row per pathway in the collection and three columns of *p*-values for each pathway.

In this package, we provide an example of SNP, DNA methylation, and gene expression results for 640 subjects with Alzheimer’s disease (from the ROSMAP study) evaluated over the Broad Institute’s C2 CP collection. We show the first 10 rows (of the 2833 pathways in the collection with fewer than 200 genes or more than 4 genes):


```{r, show_data, echo=FALSE, results='asis'}
alzheimersMultiOmics_df %>% 
	select(pathway, ends_with("Pval")) %>% 
	arrange(pathway) %>% 
	slice(1:10) %>% 
	kableExtra::kable()
```

------------------------------------------------------------------------

</br>



# The MiniMax Statistic and its Distribution


## The MiniMax statistic
From a systems biology perspective, when we perform a multi-omics analysis, we often care about biological processes which are dysregulated across *multiple* layers. For example, a moderate cascading effect from DNA methylation upstream through gene expression to protein expression may be of more practical significance than a heavily dysregulated effect in a single layer of the process. The latter scenario will be easily and quickly detected in single-omics analysis, but the former scenario will be missed.

Therefore, the pathway MiniMax statistic is defined to be the **min**imum *p*-value among all pairwise **max**ima for that biological pathway. Biologically, it indicates if the pathway is dysregulated in two or more layers of genomic data. Mathematically, such a measure is equivalent to requiring that the second smallest *p*-value is significant. That is, we can equivalently define the MiniMax statistic to be the second-order statistic of the *p*-values.


## The Distribution of the MiniMax statistic
Let us consider the scenario wherein we test that a single pathway is dysregulated for a single data platform $g \in 1, 2, \ldots, G$. Under the null hypothesis $H_0$ (that there is no signal within that pathway for genomics data set $g$), the $p$-value of a well-defined statistical test for this question follows a uniform distribution; that is, $p_g \sim U[0,1]$. Further, consider $G$ such tests for that pathway, and for the sake of argument, assume that they are independent (a horribly inaccurate assumption to be sure); then under the null hypothesis $H_0$,
$$
p_g \overset{i.i.d.}{\sim} U[0,1],\ g = 1, 2, \ldots, G.
$$
It is a known result that the order statistics for such a collection of uniform random variables follows a Beta distribution (denoted $\mathbb{B}(\alpha, \beta)$ herein). Thus, $$
\text{MiniMax}(p_1, p_2, \ldots, p_G) \equiv p_{[2]} \sim \mathbb{B}(2, G + 1 - 2).
$$


## Adjusting the parameters for dependence across data platforms
We know that this assumption of independence may not hold under real data. We believe that the pathway activity for one layer of genomic data should be related to the activity for that pathway in another genomic data layer. However, we believe that the use of the Beta distribution itself will still be appropriate, but that the parameters need to be adjusted to account for the dependence between $p_1, p_2, \ldots, p_G$.

In order to estimate these Beta distribution parameters, we recommend that you run your single-platform analyses twice: once with the real data, and once with a random permutation of phenotypes in that dataset (to simulate pathway conditions under $H_0$). Once this has been completed, pass the MiniMax statistics calculated from platform $p$-values for each pathway under $H_0$ to the `MiniMax_estBetaParams()` function. This function has options to use either the Method of Moments or Maximum Likelihood to estimate the parameters of this Beta distribution. If you do not have access to the original data (such as in the case of a meta-analysis), then the only option we offer right now is to use the closed-form definitions of these parameters ($\{\alpha = 2, \beta = G + 1 - 2\}$).


## Continuing the Alzheimer's disease example
In the data set above, the MiniMax statistics and corresponding $p$-values are given below. Note that we assume that $\{\alpha = 2, \beta = 2\}$ since the individual omics data *p*-values for DNA methylation and SNP were based on meta-analyses (we do not have access to the raw data to estimate the parameters as we would like). However, we verified *in silico* that these fixed values performed reasonably well. Therefore, we can add the statistic and its significance as follows:
```{r, run_MiniMax}
###  The results table  ---
alzheimersMultiOmics_df


###  Pathway Multi-Omics Significance with the MiniMax  ---
# We accept the default values of the Beta Distribution
adMiniMax_df <- 
	alzheimersMultiOmics_df %>% 
	# The MiniMax() function takes in the gene set name and the three p-value
	#   columns
	select(pathway, ends_with("Pval")) %>% 
	rename(
		SNP = snpPval, DNAm = dnamPval, RNAseq = rnaseqPval
	) %>% 
	MiniMax()

adMiniMax_df
```

Finally, we can adjust these pathway $p$-values for multiple comparisons and filter to those most significant.
```{r, adjust_results, results='asis'}
adRes_df <- 
	adMiniMax_df %>% 
	mutate(MiniMaxFDR = p.adjust(MiniMaxP, method = "fdr")) %>% 
	filter(MiniMaxFDR < 0.01) 

adRes_df %>% 
	select(pathway, MiniMaxFDR, drivers) %>% 
	mutate(pathway = str_trunc(pathway, width = 45)) %>% 
	kableExtra::kable()
```

In our manuscript, we demonstrate many of the top pathways have been implicated in Alzheimer’s disease or in neurodegenerative conditions at large.

------------------------------------------------------------------------

</br>



# Session Information

```{r}
sessionInfo()
```
