<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Using the MiniMax Statistic • pathwayMultiomics</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Using the MiniMax Statistic">
<meta property="og:description" content="pathwayMultiomics">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">pathwayMultiomics</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9004</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/intro_to_minimax.html">Using the MiniMax Statistic</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/TransBioInfoLab/pathwayMultiomics/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="intro_to_minimax_files/header-attrs-2.10/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Using the MiniMax Statistic</h1>
                        <h4 class="author">Gabriel Odom and Lily Wang</h4>
            
            <h4 class="date">8/29/2021</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/TransBioInfoLab/pathwayMultiomics/blob/master/vignettes/intro_to_minimax.Rmd"><code>vignettes/intro_to_minimax.Rmd</code></a></small>
      <div class="hidden name"><code>intro_to_minimax.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>This vignette will give a brief overview of the practical problem that motivated us to work on this project, some mathematical details about our method, and a walkthrough of a real data analysis example.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org">tidyverse</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/TransBioInfoLab/pathwayMultiomics">pathwayMultiomics</a></span><span class="op">)</span></code></pre></div>
<hr>
<p></p>
</div>
<div id="motivating-example" class="section level1">
<h1 class="hasAnchor">
<a href="#motivating-example" class="anchor"></a>Motivating Example</h1>
<p>Some time ago, our research team attempted to use multi-omics tools to answer a few questions concerning colorectal cancer. While the tools were often well-documented and had nice statistical properties, we consistently ran into one of the following two constraints:</p>
<ul>
<li>Multi-Omics tools must match across platforms <em>by samples</em>, or</li>
<li>Multi-Omics tools must match across platforms <em>by features</em> (e.g. genes).</li>
</ul>
<div id="joining-data-platforms" class="section level2">
<h2 class="hasAnchor">
<a href="#joining-data-platforms" class="anchor"></a>Joining data platforms</h2>
<p>In many cancer data sets, we could usually find a nice overlapping set of samples between two platforms of data (for instance, copy number variation and DNA methylation). However, as soon as we added in a third or fourth platform (such as gene and/or protein expression), the intersection of the samples recorded with these new platforms rapidly approached the empty set. We could start with 600+ samples, and we’d barely have 50 samples by the time we added a third platform and only a handful of samples (or even none at all) if we added a fourth platform.</p>
<p>If instead, we tried to match on features, we would run into a different set of issues. What is the best way to map from the “tallest” data, site- or probe-level data (as in DNA methylation, SNP, or RNA editing levels) to the “shortest” data, such as expressions of small sets of individual proteins? Using a feature-matching strategy would require us to discard many relevant biological features simply because they were not recorded in the most restricted data.</p>
</div>
<div id="a-pathway-based-solution" class="section level2">
<h2 class="hasAnchor">
<a href="#a-pathway-based-solution" class="anchor"></a>A Pathway-based solution</h2>
<p>Rather than using only the intersection of the samples or only the intersection of the genomic features, we proposed to match across data sets by data summaries of biological pathway <em>p</em>-values. For example, consider omics data measured on three data platforms for colorectal adenocarcinoma: copy number variation, DNA methylation, and protein expression. These three data sets may share samples, but that isn’t required.</p>
<p>Our first step is to statistically assess the activity of a set of biological pathways in each genomic dataset independently and then match the pathway <em>p</em>-values across the three datasets. The resulting summary data would have one row per pathway in the collection and three columns of <em>p</em>-values for each pathway.</p>
<p>In this package, we provide an example of SNP, DNA methylation, and gene expression results for 640 subjects with Alzheimer’s disease (from the ROSMAP study) evaluated over the Broad Institute’s C2 CP collection. We show the first 10 rows (of the 2833 pathways in the collection with fewer than 200 genes or more than 4 genes):</p>
<table class="table">
<thead><tr>
<th style="text-align:left;">
pathway
</th>
<th style="text-align:right;">
snpPval
</th>
<th style="text-align:right;">
dnamPval
</th>
<th style="text-align:right;">
rnaseqPval
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
BIOCARTA_41BB_PATHWAY
</td>
<td style="text-align:right;">
0.7792572
</td>
<td style="text-align:right;">
0.8163709
</td>
<td style="text-align:right;">
0.0371308
</td>
</tr>
<tr>
<td style="text-align:left;">
BIOCARTA_ACE2_PATHWAY
</td>
<td style="text-align:right;">
0.8778045
</td>
<td style="text-align:right;">
0.2253418
</td>
<td style="text-align:right;">
0.0011873
</td>
</tr>
<tr>
<td style="text-align:left;">
BIOCARTA_ACETAMINOPHEN_PATHWAY
</td>
<td style="text-align:right;">
0.2675038
</td>
<td style="text-align:right;">
0.2972964
</td>
<td style="text-align:right;">
0.6133866
</td>
</tr>
<tr>
<td style="text-align:left;">
BIOCARTA_ACH_PATHWAY
</td>
<td style="text-align:right;">
0.5626200
</td>
<td style="text-align:right;">
0.5347566
</td>
<td style="text-align:right;">
0.0389610
</td>
</tr>
<tr>
<td style="text-align:left;">
BIOCARTA_ACTINY_PATHWAY
</td>
<td style="text-align:right;">
0.4538971
</td>
<td style="text-align:right;">
0.1351410
</td>
<td style="text-align:right;">
0.0359640
</td>
</tr>
<tr>
<td style="text-align:left;">
BIOCARTA_AGPCR_PATHWAY
</td>
<td style="text-align:right;">
0.5998621
</td>
<td style="text-align:right;">
0.0118939
</td>
<td style="text-align:right;">
0.0187746
</td>
</tr>
<tr>
<td style="text-align:left;">
BIOCARTA_AGR_PATHWAY
</td>
<td style="text-align:right;">
0.4561920
</td>
<td style="text-align:right;">
0.9892614
</td>
<td style="text-align:right;">
0.2667333
</td>
</tr>
<tr>
<td style="text-align:left;">
BIOCARTA_AHSP_PATHWAY
</td>
<td style="text-align:right;">
0.7556747
</td>
<td style="text-align:right;">
0.4608289
</td>
<td style="text-align:right;">
0.2947053
</td>
</tr>
<tr>
<td style="text-align:left;">
BIOCARTA_AKAP13_PATHWAY
</td>
<td style="text-align:right;">
0.8725796
</td>
<td style="text-align:right;">
0.3237901
</td>
<td style="text-align:right;">
0.2807193
</td>
</tr>
<tr>
<td style="text-align:left;">
BIOCARTA_AKAP95_PATHWAY
</td>
<td style="text-align:right;">
0.7723686
</td>
<td style="text-align:right;">
0.2584549
</td>
<td style="text-align:right;">
0.0369630
</td>
</tr>
</tbody>
</table>
<hr>
<p></p>
</div>
</div>
<div id="the-minimax-statistic-and-its-distribution" class="section level1">
<h1 class="hasAnchor">
<a href="#the-minimax-statistic-and-its-distribution" class="anchor"></a>The MiniMax Statistic and its Distribution</h1>
<div id="the-minimax-statistic" class="section level2">
<h2 class="hasAnchor">
<a href="#the-minimax-statistic" class="anchor"></a>The MiniMax statistic</h2>
<p>From a systems biology perspective, when we perform a multi-omics analysis, we often care about biological processes which are dysregulated across <em>multiple</em> layers. For example, a moderate cascading effect from DNA methylation upstream through gene expression to protein expression may be of more practical significance than a heavily dysregulated effect in a single layer of the process. The latter scenario will be easily and quickly detected in single-omics analysis, but the former scenario will be missed.</p>
<p>Therefore, the pathway MiniMax statistic is defined to be the <strong>min</strong>imum <em>p</em>-value among all pairwise <strong>max</strong>ima for that biological pathway. Biologically, it indicates if the pathway is dysregulated in two or more layers of genomic data. Mathematically, such a measure is equivalent to requiring that the second smallest <em>p</em>-value is significant. That is, we can equivalently define the MiniMax statistic to be the second-order statistic of the <em>p</em>-values.</p>
</div>
<div id="the-distribution-of-the-minimax-statistic" class="section level2">
<h2 class="hasAnchor">
<a href="#the-distribution-of-the-minimax-statistic" class="anchor"></a>The Distribution of the MiniMax statistic</h2>
<p>Let us consider the scenario wherein we test that a single pathway is dysregulated for a single data platform <span class="math inline">\(g \in 1, 2, \ldots, G\)</span>. Under the null hypothesis <span class="math inline">\(H_0\)</span> (that there is no signal within that pathway for genomics data set <span class="math inline">\(g\)</span>), the <span class="math inline">\(p\)</span>-value of a well-defined statistical test for this question follows a uniform distribution; that is, <span class="math inline">\(p_g \sim U[0,1]\)</span>. Further, consider <span class="math inline">\(G\)</span> such tests for that pathway, and for the sake of argument, assume that they are independent (a horribly inaccurate assumption to be sure); then under the null hypothesis <span class="math inline">\(H_0\)</span>, <span class="math display">\[
p_g \overset{i.i.d.}{\sim} U[0,1],\ g = 1, 2, \ldots, G.
\]</span> It is a known result that the order statistics for such a collection of uniform random variables follows a Beta distribution (denoted <span class="math inline">\(\mathbb{B}(\alpha, \beta)\)</span> herein). Thus, <span class="math display">\[
\text{MiniMax}(p_1, p_2, \ldots, p_G) \equiv p_{[2]} \sim \mathbb{B}(2, G + 1 - 2).
\]</span></p>
</div>
<div id="adjusting-the-parameters-for-dependence-across-data-platforms" class="section level2">
<h2 class="hasAnchor">
<a href="#adjusting-the-parameters-for-dependence-across-data-platforms" class="anchor"></a>Adjusting the parameters for dependence across data platforms</h2>
<p>We know that this assumption of independence may not hold under real data. We believe that the pathway activity for one layer of genomic data should be related to the activity for that pathway in another genomic data layer. However, we believe that the use of the Beta distribution itself will still be appropriate, but that the parameters need to be adjusted to account for the dependence between <span class="math inline">\(p_1, p_2, \ldots, p_G\)</span>.</p>
<p>In order to estimate these Beta distribution parameters, we recommend that you run your single-platform analyses twice: once with the real data, and once with a random permutation of phenotypes in that dataset (to simulate pathway conditions under <span class="math inline">\(H_0\)</span>). Once this has been completed, pass the MiniMax statistics calculated from platform <span class="math inline">\(p\)</span>-values for each pathway under <span class="math inline">\(H_0\)</span> to the <code><a href="../reference/MiniMax_estBetaParams.html">MiniMax_estBetaParams()</a></code> function. This function has options to use either the Method of Moments or Maximum Likelihood to estimate the parameters of this Beta distribution. If you do not have access to the original data (such as in the case of a meta-analysis), then the only option we offer right now is to use the closed-form definitions of these parameters (<span class="math inline">\(\{\alpha = 2, \beta = G + 1 - 2\}\)</span>).</p>
</div>
<div id="continuing-the-alzheimers-disease-example" class="section level2">
<h2 class="hasAnchor">
<a href="#continuing-the-alzheimers-disease-example" class="anchor"></a>Continuing the Alzheimer’s disease example</h2>
<p>In the data set above, the MiniMax statistics and corresponding <span class="math inline">\(p\)</span>-values are given below. Note that we assume that <span class="math inline">\(\{\alpha = 2, \beta = 2\}\)</span> since the individual omics data <em>p</em>-values for DNA methylation and SNP were based on meta-analyses (we do not have access to the raw data to estimate the parameters as we would like). However, we verified <em>in silico</em> that these fixed values performed reasonably well. Therefore, we can add the statistic and its significance as follows:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">###  The results table  ---</span>
<span class="va">alzheimersMultiOmics_df</span>
<span class="co">#&gt; # A tibble: 2,833 × 8</span>
<span class="co">#&gt;    pathway          size  snpPval   snpFDR dnamPval dnamFDR rnaseqPval rnaseqFDR</span>
<span class="co">#&gt;    &lt;chr&gt;           &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt;   &lt;dbl&gt;      &lt;dbl&gt;     &lt;dbl&gt;</span>
<span class="co">#&gt;  1 BIOCARTA_GRANU…    14 9.67e- 1 9.99e- 1  0.00241   0.359     0.919     0.936 </span>
<span class="co">#&gt;  2 BIOCARTA_LYM_P…    11 6.79e- 1 9.99e- 1  0.0156    0.538     0.0749    0.158 </span>
<span class="co">#&gt;  3 BIOCARTA_BLYMP…    12 3.31e-69 1.17e-66  0.138     0.858     0.999     0.999 </span>
<span class="co">#&gt;  4 BIOCARTA_CARM_…    24 3.30e- 1 9.99e- 1  0.0263    0.643     0.0287    0.0878</span>
<span class="co">#&gt;  5 BIOCARTA_LAIR_…    17 2.17e- 1 9.99e- 1  0.0463    0.718     0.630     0.700 </span>
<span class="co">#&gt;  6 BIOCARTA_VDR_P…    24 8.78e- 1 9.99e- 1  0.115     0.840     0.0350    0.0988</span>
<span class="co">#&gt;  7 BIOCARTA_MTA3_…    18 7.98e- 1 9.99e- 1  0.0494    0.721     0.105     0.197 </span>
<span class="co">#&gt;  8 BIOCARTA_GABA_…     9 3.71e- 1 9.99e- 1  1         1         0.0280    0.0865</span>
<span class="co">#&gt;  9 BIOCARTA_EGFR_…    11 6.77e- 1 9.99e- 1  0.0339    0.668     0.118     0.213 </span>
<span class="co">#&gt; 10 BIOCARTA_MONOC…    10 5.26e- 1 9.99e- 1  0.657     1         0.167     0.268 </span>
<span class="co">#&gt; # … with 2,823 more rows</span>


<span class="co">###  Pathway Multi-Omics Significance with the MiniMax  ---</span>
<span class="co"># We accept the default values of the Beta Distribution</span>
<span class="va">adMiniMax_df</span> <span class="op">&lt;-</span> 
    <span class="va">alzheimersMultiOmics_df</span> <span class="op">%&gt;%</span> 
    <span class="co"># The MiniMax() function takes in the gene set name and the three p-value</span>
    <span class="co">#   columns</span>
    <span class="fu">select</span><span class="op">(</span><span class="va">pathway</span>, <span class="fu">ends_with</span><span class="op">(</span><span class="st">"Pval"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> 
    <span class="fu">rename</span><span class="op">(</span>
        SNP <span class="op">=</span> <span class="va">snpPval</span>, DNAm <span class="op">=</span> <span class="va">dnamPval</span>, RNAseq <span class="op">=</span> <span class="va">rnaseqPval</span>
    <span class="op">)</span> <span class="op">%&gt;%</span> 
    <span class="fu"><a href="../reference/MiniMax.html">MiniMax</a></span><span class="op">(</span><span class="op">)</span>

<span class="va">adMiniMax_df</span>
<span class="co">#&gt; # A tibble: 2,833 × 7</span>
<span class="co">#&gt;    pathway                     SNP     DNAm   RNAseq MiniMax  MiniMaxP drivers  </span>
<span class="co">#&gt;    &lt;chr&gt;                     &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt;   &lt;dbl&gt;     &lt;dbl&gt; &lt;chr&gt;    </span>
<span class="co">#&gt;  1 PID_PDGFRB_PATHWAY     6.99e- 1 0.000145  1.67e-4 1.67e-4   8.33e-8 DNAm and…</span>
<span class="co">#&gt;  2 WP_CHEMOKINE_SIGNALIN… 8.19e- 1 0.000317  1.94e-5 3.17e-4   3.02e-7 DNAm and…</span>
<span class="co">#&gt;  3 KEGG_HEMATOPOIETIC_CE… 3.67e-36 0.000340  7.61e-1 3.40e-4   3.48e-7 DNAm and…</span>
<span class="co">#&gt;  4 PID_TCR_PATHWAY        4.48e- 4 0.0275    4.90e-4 4.90e-4   7.20e-7 RNAseq a…</span>
<span class="co">#&gt;  5 WP_REGULATION_OF_TOLL… 3.76e- 5 0.0332    6.55e-4 6.55e-4   1.29e-6 RNAseq a…</span>
<span class="co">#&gt;  6 KEGG_CHEMOKINE_SIGNAL… 7.90e- 1 0.000672  2.98e-4 6.72e-4   1.35e-6 DNAm and…</span>
<span class="co">#&gt;  7 PID_KIT_PATHWAY        2.55e- 1 0.000684  1.10e-4 6.84e-4   1.40e-6 DNAm and…</span>
<span class="co">#&gt;  8 WP_KIT_RECEPTOR_SIGNA… 3.05e- 2 0.000368  1.41e-3 1.41e-3   5.94e-6 DNAm and…</span>
<span class="co">#&gt;  9 PID_CXCR4_PATHWAY      4.87e- 4 0.00150   7.29e-2 1.50e-3   6.76e-6 DNAm and…</span>
<span class="co">#&gt; 10 REACTOME_TCR_SIGNALING 6.06e-52 0.207     2.16e-3 2.16e-3   1.40e-5 RNAseq a…</span>
<span class="co">#&gt; # … with 2,823 more rows</span></code></pre></div>
<p>Finally, we can adjust these pathway <span class="math inline">\(p\)</span>-values for multiple comparisons and filter to those most significant.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">adRes_df</span> <span class="op">&lt;-</span> 
    <span class="va">adMiniMax_df</span> <span class="op">%&gt;%</span> 
    <span class="fu">mutate</span><span class="op">(</span>MiniMaxFDR <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html">p.adjust</a></span><span class="op">(</span><span class="va">MiniMaxP</span>, method <span class="op">=</span> <span class="st">"fdr"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> 
    <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">MiniMaxFDR</span> <span class="op">&lt;</span> <span class="fl">0.01</span><span class="op">)</span> 

<span class="va">adRes_df</span> <span class="op">%&gt;%</span> 
    <span class="fu">select</span><span class="op">(</span><span class="va">pathway</span>, <span class="va">MiniMaxFDR</span>, <span class="va">drivers</span><span class="op">)</span> <span class="op">%&gt;%</span> 
    <span class="fu">mutate</span><span class="op">(</span>pathway <span class="op">=</span> <span class="fu">str_trunc</span><span class="op">(</span><span class="va">pathway</span>, width <span class="op">=</span> <span class="fl">45</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> 
    <span class="fu">kableExtra</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<table class="table">
<thead><tr>
<th style="text-align:left;">
pathway
</th>
<th style="text-align:right;">
MiniMaxFDR
</th>
<th style="text-align:left;">
drivers
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
PID_PDGFRB_PATHWAY
</td>
<td style="text-align:right;">
0.0002360
</td>
<td style="text-align:left;">
DNAm and RNAseq
</td>
</tr>
<tr>
<td style="text-align:left;">
WP_CHEMOKINE_SIGNALING_PATHWAY
</td>
<td style="text-align:right;">
0.0003282
</td>
<td style="text-align:left;">
DNAm and RNAseq
</td>
</tr>
<tr>
<td style="text-align:left;">
KEGG_HEMATOPOIETIC_CELL_LINEAGE
</td>
<td style="text-align:right;">
0.0003282
</td>
<td style="text-align:left;">
DNAm and SNP
</td>
</tr>
<tr>
<td style="text-align:left;">
PID_TCR_PATHWAY
</td>
<td style="text-align:right;">
0.0005100
</td>
<td style="text-align:left;">
RNAseq and SNP
</td>
</tr>
<tr>
<td style="text-align:left;">
WP_REGULATION_OF_TOLLLIKE_RECEPTOR_SIGNALI…
</td>
<td style="text-align:right;">
0.0005686
</td>
<td style="text-align:left;">
RNAseq and SNP
</td>
</tr>
<tr>
<td style="text-align:left;">
KEGG_CHEMOKINE_SIGNALING_PATHWAY
</td>
<td style="text-align:right;">
0.0005686
</td>
<td style="text-align:left;">
DNAm and RNAseq
</td>
</tr>
<tr>
<td style="text-align:left;">
PID_KIT_PATHWAY
</td>
<td style="text-align:right;">
0.0005686
</td>
<td style="text-align:left;">
DNAm and RNAseq
</td>
</tr>
<tr>
<td style="text-align:left;">
WP_KIT_RECEPTOR_SIGNALING_PATHWAY
</td>
<td style="text-align:right;">
0.0021048
</td>
<td style="text-align:left;">
DNAm and RNAseq
</td>
</tr>
<tr>
<td style="text-align:left;">
PID_CXCR4_PATHWAY
</td>
<td style="text-align:right;">
0.0021267
</td>
<td style="text-align:left;">
DNAm and SNP
</td>
</tr>
<tr>
<td style="text-align:left;">
REACTOME_TCR_SIGNALING
</td>
<td style="text-align:right;">
0.0039757
</td>
<td style="text-align:left;">
RNAseq and SNP
</td>
</tr>
<tr>
<td style="text-align:left;">
REACTOME_HATS_ACETYLATE_HISTONES
</td>
<td style="text-align:right;">
0.0071556
</td>
<td style="text-align:left;">
RNAseq and SNP
</td>
</tr>
<tr>
<td style="text-align:left;">
REACTOME_MHC_CLASS_II_ANTIGEN_PRESENTATION
</td>
<td style="text-align:right;">
0.0071556
</td>
<td style="text-align:left;">
RNAseq and SNP
</td>
</tr>
<tr>
<td style="text-align:left;">
WP_INSULIN_SIGNALING
</td>
<td style="text-align:right;">
0.0071556
</td>
<td style="text-align:left;">
DNAm and RNAseq
</td>
</tr>
<tr>
<td style="text-align:left;">
REACTOME_RAC1_GTPASE_CYCLE
</td>
<td style="text-align:right;">
0.0076145
</td>
<td style="text-align:left;">
DNAm and RNAseq
</td>
</tr>
<tr>
<td style="text-align:left;">
REACTOME_DDX58_IFIH1_MEDIATED_INDUCTION_OF…
</td>
<td style="text-align:right;">
0.0076145
</td>
<td style="text-align:left;">
RNAseq and SNP
</td>
</tr>
<tr>
<td style="text-align:left;">
KEGG_NEUROTROPHIN_SIGNALING_PATHWAY
</td>
<td style="text-align:right;">
0.0076145
</td>
<td style="text-align:left;">
DNAm and RNAseq
</td>
</tr>
<tr>
<td style="text-align:left;">
WP_TRANSCRIPTION_FACTOR_REGULATION_IN_ADIP…
</td>
<td style="text-align:right;">
0.0077424
</td>
<td style="text-align:left;">
DNAm and RNAseq
</td>
</tr>
<tr>
<td style="text-align:left;">
REACTOME_SUMOYLATION_OF_CHROMATIN_ORGANIZA…
</td>
<td style="text-align:right;">
0.0094065
</td>
<td style="text-align:left;">
RNAseq and SNP
</td>
</tr>
<tr>
<td style="text-align:left;">
REACTOME_HCMV_INFECTION
</td>
<td style="text-align:right;">
0.0094065
</td>
<td style="text-align:left;">
RNAseq and SNP
</td>
</tr>
<tr>
<td style="text-align:left;">
REACTOME_NEUROTRANSMITTER_RECEPTORS_AND_PO…
</td>
<td style="text-align:right;">
0.0097309
</td>
<td style="text-align:left;">
RNAseq and SNP
</td>
</tr>
</tbody>
</table>
<p>In our manuscript, we demonstrate many of the top pathways have been implicated in Alzheimer’s disease or in neurodegenerative conditions at large.</p>
<hr>
<p></p>
</div>
</div>
<div id="session-information" class="section level1">
<h1 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session Information</h1>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R version 4.1.0 (2021-05-18)</span>
<span class="co">#&gt; Platform: x86_64-apple-darwin17.0 (64-bit)</span>
<span class="co">#&gt; Running under: macOS Big Sur 10.16</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib</span>
<span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt;  [1] pathwayMultiomics_0.0.0.9004 forcats_0.5.1               </span>
<span class="co">#&gt;  [3] stringr_1.4.0                dplyr_1.0.7                 </span>
<span class="co">#&gt;  [5] purrr_0.3.4                  readr_2.0.0                 </span>
<span class="co">#&gt;  [7] tidyr_1.1.3                  tibble_3.1.3                </span>
<span class="co">#&gt;  [9] ggplot2_3.3.5                tidyverse_1.3.1             </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;  [1] Rcpp_1.0.7         svglite_2.0.0      lubridate_1.7.10   assertthat_0.2.1  </span>
<span class="co">#&gt;  [5] rprojroot_2.0.2    digest_0.6.27      utf8_1.2.2         R6_2.5.0          </span>
<span class="co">#&gt;  [9] cellranger_1.1.0   backports_1.2.1    RcppZiggurat_0.1.6 reprex_2.0.1      </span>
<span class="co">#&gt; [13] evaluate_0.14      highr_0.9          httr_1.4.2         pillar_1.6.2      </span>
<span class="co">#&gt; [17] rlang_0.4.11       readxl_1.3.1       rstudioapi_0.13    jquerylib_0.1.4   </span>
<span class="co">#&gt; [21] rmarkdown_2.10     pkgdown_1.6.1      textshaping_0.3.5  desc_1.3.0        </span>
<span class="co">#&gt; [25] webshot_0.5.2      munsell_0.5.0      broom_0.7.9        compiler_4.1.0    </span>
<span class="co">#&gt; [29] modelr_0.1.8       xfun_0.25          pkgconfig_2.0.3    systemfonts_1.0.2 </span>
<span class="co">#&gt; [33] htmltools_0.5.1.1  tidyselect_1.1.1   viridisLite_0.4.0  fansi_0.5.0       </span>
<span class="co">#&gt; [37] crayon_1.4.1       tzdb_0.1.2         dbplyr_2.1.1       withr_2.4.2       </span>
<span class="co">#&gt; [41] grid_4.1.0         jsonlite_1.7.2     gtable_0.3.0       lifecycle_1.0.0   </span>
<span class="co">#&gt; [45] DBI_1.1.1          magrittr_2.0.1     scales_1.1.1       Rfast_2.0.3       </span>
<span class="co">#&gt; [49] cli_3.0.1          stringi_1.7.3      cachem_1.0.5       fs_1.5.0          </span>
<span class="co">#&gt; [53] xml2_1.3.2         bslib_0.2.5.1      ellipsis_0.3.2     ragg_1.1.3        </span>
<span class="co">#&gt; [57] generics_0.1.0     vctrs_0.3.8        kableExtra_1.3.4   tools_4.1.0       </span>
<span class="co">#&gt; [61] glue_1.4.2         hms_1.1.0          parallel_4.1.0     fastmap_1.1.0     </span>
<span class="co">#&gt; [65] yaml_2.2.1         colorspace_2.0-2   rvest_1.0.1        memoise_2.0.0     </span>
<span class="co">#&gt; [69] knitr_1.33         haven_2.4.3        sass_0.4.0</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Gabriel Odom, Lily Wang.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
